### PR TITLE
RDK-35406 : XCAST unregisterApplications API

### PR DESCRIPTION
**Pull request by acheri988**
Reason for change:
XCAST unregisterApplications API
Test Procedure: None
Risks: Low

Change-Id: Ifa45888f90c2490e9df9ba6230b90e6d2437a2db Signed-off-by:Anooj Cheriyan <Anooj_Cheriyan@comcast.com>

### CODE REVIEW

File: XCast/RtXcastConnector.cpp [hunk 1]
```diff
@@ -398,199 +351,47 @@ string RtXcastConnector::getProtocolVersion(void)
...
-            LOGINFO(" xdialCastObj is NULL ");
+        appReqList->pushBack(rtValue(appReq));
     }
-}
 
+    if(xdialCastObj != NULL)
+    {
+        LOGINFO("%s:%d xdialCastObj Not NULL", __FUNCTION__, __LINE__);
+        int ret = xdialCastObj.send("onRegisterApplications", appReqList);
+        LOGINFO("XcastService send onRegisterApplications ret:%d",ret);
```
Comment (apatel859) [inline XCast/RtXcastConnector.cpp:390]: combine 388 here

File: docs/api/XCastPlugin.md [hunk 2]
```diff
@@ -378,8 +379,8 @@ Registers an application. This allows to whitelist the apps which support dial s
 | result | object |  |
 | result.success | boolean | Whether the request succeeded |
 
-### Example
-
+### Example 1
+Following call will register NetflixApp to the gdisl dynamic list.
```
Comment (apatel859) [inline docs/api/XCastPlugin.md:383]: dial

File: docs/api/XCastPlugin.md [hunk 3]
```diff
@@ -388,7 +389,122 @@ Registers an application. This allows to whitelist the apps which support dial s
...
+{
+    "jsonrpc": "2.0",
+    "id": 42,
+    "result": {
+        "success": true
+    }
+}
+```
+### Example 2
+Following call will update the existing NetflixApp properties to the gdisl dynamic list and insert new applications YouTube, YouTubeKids and YouTubeTV.
```
Comment (apatel859) [inline docs/api/XCastPlugin.md:412]: dial

File: XCast/XCast.cpp [hunk 4]
```diff
@@ -356,6 +353,305 @@ uint32_t XCast::getProtocolVersion(const JsonObject& parameters, JsonObject& res
...
+                }
+                break;
+            }
+        }
+    }
+    return isEntryFound;
+}
+
+void XCast::dumpDynamicAppConfigCache(string strListName, std::vector<DynamicAppConfig*> appConfigList) {
+    LOGINFO("appConfigList count: %d", appConfigList.size());
```
Comment (apatel859) [inline XCast/XCast.cpp:379]: combine with 382

File: XCast/XCast.cpp [hunk 5]
```diff
@@ -356,6 +353,305 @@ uint32_t XCast::getProtocolVersion(const JsonObject& parameters, JsonObject& res
...
+                }
+
+            }
+            for (DynamicAppConfig* pDynamicAppConfig : appConfigListTemp) {
+                appConfigList.push_back(pDynamicAppConfig);
+            }
+            appConfigListTemp.clear();
+            iIndex++;
+        }
+        LOGINFO("");
```
Comment (apatel859) [inline XCast/XCast.cpp:630]: remove

File: docs/api/XCastPlugin.md [hunk 6]
```diff
@@ -358,7 +359,7 @@ The following table provides a client error mapping example:
 <a name="registerApplications"></a>
 ## *registerApplications*
 
-Registers an application. This allows to whitelist the apps which support dial service. To dynamically update the app list, same API should be called with the updated list.
+This API registers applications passed with XCast service. This allows to whitelist the apps which support dial service. Calling this api again, with a new set of APP names, will append those APP names to existing whitelist. Passing an existing whitelisted app, with a modified parameter value, will update the curresponding field in the whitelist.
```
Comment (apatel859) [inline docs/api/XCastPlugin.md:362]: This API allows to application to whitelist the apps which support dial service with xcast service. Application can register single or multiple apps by passing application list.

**Review by apatel859 — APPROVED** [review]

**Comment by acheri988** [comment]
New one created: https://github.com/rdkcentral/rdkservices/pull/3194

**Comment by anand-ky** [comment]
Closing older PRs. Pls bring changes to latest sprint.

**Pull request closed without merging**